### PR TITLE
[VariableDistance] Experimental MediaQuery expansion

### PR DESCRIPTION
**Pull request by willlarche**

### CODE REVIEW

File: variable_distance_exploration/lib/experimental_media_query.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  /// Default is [DeviceCorporealContext.focused].
+  final DeviceCorporealContext deviceCorporealContext;
+
+  /// Intended purpose of the screen in the real world.
+  ///
+  /// Default is [DeviceCorporealType.slabHandheld].
+  final DeviceCorporealType deviceCorporealType;
+
+  /// Number of users facing the screen.
+  /// Null means 'unknown'.
```
Comment (johnsonmh) [inline variable_distance_exploration/lib/experimental_media_query.dart]: Dart Nit: `/// Null means 'unknown'.` should be on a new line. 

https://www.dartlang.org/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

File: variable_distance_exploration/lib/experimental_media_query.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  ///
+  /// Default is null.
+  final int viewerCount;
+
+  /// If the phone is foldable, is it folded (true) or not (false).
+  /// If the phone is not foldable, null.
```
Comment (johnsonmh) [inline variable_distance_exploration/lib/experimental_media_query.dart]: Dart nit: `/// If the phone is not foldable, null.` should be on a new line.

https://www.dartlang.org/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

File: variable_distance_exploration/lib/experimental_media_query.dart [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  desktop,
+
+  /// Foldable phone.
+  foldableHandheld,
+
+  /// Slab-style phone.
+  slabHandheld,
+
+  /// Mobile tablet.
+  tabletHandheld,
```
Comment (johnsonmh) [inline variable_distance_exploration/lib/experimental_media_query.dart]: I don't know much about the conversation going on here, but it seems reasonable to me to maybe consider `tabletHandheld`s to just be the same as `slabHandheld`s, just differing in size. They seem to have the same general form.

Just a thought.

Comment (willlarche) [reply variable_distance_exploration/lib/experimental_media_query.dart]: They differ in layout expectations. Landscape tablet tends to have a split screen.

File: variable_distance_exploration/lib/experimental_media_query.dart (hunk 2 above)
Comment (willlarche) [reply variable_distance_exploration/lib/experimental_media_query.dart]: Done.

File: variable_distance_exploration/lib/experimental_media_query.dart (hunk 1 above)
Comment (willlarche) [reply variable_distance_exploration/lib/experimental_media_query.dart]: Done.

File: variable_distance_exploration/lib/experimental_media_query.dart [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+/// Interaction models.
+enum DeviceCorporealContext {
+  /// Viewed only.
+  ambient,
+
+  /// Interactive.
```
Comment (clocksmith) [inline variable_distance_exploration/lib/experimental_media_query.dart]: why isn't this field called interactive?

File: variable_distance_exploration/lib/experimental_media_query.dart [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  final int viewerCount;
+
+  /// If the phone is foldable, is it folded (true) or not (false).
+  /// 
+  /// If the phone is not foldable, null.
+  ///
+  /// Default is null.
+  final bool isFolded;
+
+  /// Angle of eyes of the main user to the screen.
```
Comment (clocksmith) [inline variable_distance_exploration/lib/experimental_media_query.dart]: Is this the vertical angle (altitude)? Distinguish between horizontal, azimuth, etc

Comment (willlarche) [reply variable_distance_exploration/lib/experimental_media_query.dart]: Altitude Angle. Do you think that's clearer?

File: variable_distance_exploration/lib/experimental_media_query.dart (hunk 4 above)
Comment (willlarche) [reply variable_distance_exploration/lib/experimental_media_query.dart]: We're working on refining the vocabulary. This is what the designers were using at the moment tho. So I aligned with them until we agree on more permanent vocab.

File: variable_distance_exploration/lib/experimental_media_query.dart (hunk 5 above)
Comment (willlarche) [reply variable_distance_exploration/lib/experimental_media_query.dart]: Viewing angle. There's an industry term.

Comment (willlarche) [reply variable_distance_exploration/lib/experimental_media_query.dart]: But I updated the comments to use your terms too.

File: variable_distance_exploration/lib/experimental_media_query.dart [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  final DeviceCorporealType deviceCorporealType;
+
+  /// Number of users facing the screen.
+  ///
+  /// Null means 'unknown'.
+  ///
+  /// Default is null.
+  final int viewerCount;
+
+  /// If the phone is foldable, is it folded (true) or not (false).
```
[the remark this answers is not in the document]
Comment (rami-a) [reply variable_distance_exploration/lib/experimental_media_query.dart]: Just realized this is already merged. Oops

**Pull request merged**